adr/0006-dev-discussions.adoc

```diff
@@ -0,0 +1,125 @@
+
+A dev interested in doing contributions by following the ongoing conversations - which of the many channels do they follow? 
+
+== Considered options 
+
+Primarily use quarkus-dev mailing list::
+    Advantages::::
+        * well-known and understood by most if not all existing Quarkus contributors (~1000+ subscribers)
+        * can post, comment and read with any e-mail client
+        * filters out most noise as people need to subscribe
+    Disadvantages::::
+        * over time it has died out, rarely where conversations starts/continues being discussed
+        * e-mail/google groups is just not where new people show up as much now
+        * requires subscribing to specific list to post
+        * spam (fixed now by moderation)
+        * arbitrary blocking of mails (specifally @gsmet and others)
+
+Primarily have "Design Discussions" a sub-category of Quarkus Community/User Discussions::
+    Advantages::::
+        * Centralized location for discussions, anyone sees it
+        * Anyone can read
+        * Anyone with GitHub account can post/reply
+        * Moderation tools available to battle spam/noise
+    Disadvantages::::
+        * Notifications via email comes for *all* discussions, not possible to choose specific category
+        * User forums gets intermixed with Developer/Design discussions
+
+Primarily use issues/PR's::
+    Advantages::::
+        * Conversation directly linked to code changes
+    Disadvantages::::
+        * Not possible to just monitor the overall design discussions as all mixed up with anything else
+        * Notification filtering is quite limited
+        * Limited threading in webui
+        * No-threading on email responses
+
+Primarily use Zulip Chat::
+    Advantages::::
+        * Real-time communication
+    Disadvantages::::
+        * Only for those who have time/privilege to follow along
+
+Use private chat/mail::
+    Advantages::::
+        * Direct and focused private communication (sometimes useful)
+    Disadvantages::::
+        * Not great for open governance unless culture to comment/report on conclusion somewhere more public
+        * Cannot be located/found when trying to search for answers/why something was done
+
+Status quo/Do nothing::
+    Advantages::::
+        * Nothing need to change
+    Disadvanteges::::
+        * All of the above disadvantages
+        * No clear place to go for design discussions
+
+Seperate quarkusio/quarkus-dev repository with discussions::
+    Advantages::::
+        * Clear place to go for design discussions
+        * Easy to find and follow discussions
+        * Can filter/focus separately on user versus developer discussions
+        * Can use e-mail, at least for commenting
+        * ADR can be moved there
+        * Future call minutes can be moved there
+        * Does not break existing user conversations
+    Disadvantages::::
+        * Need to move discussions from other places to this new place
```

Comment:
Is "move" here a move of actual content, or just a move of people's habit? It would be nice to move the content so that people don't have to look in all the old places for old content, post-move, but I'm not sure of a mechanism that could do that. 
Assuming we don't migrate the content, we should add another disadvantage, which is that people either need to check the mailing list + main repo discussions to read historic discussions (increased fragmentation of 'where are discussions'), or we lose all of the historic discussions.

Reply:
I was thinking about the move of people not content here. 

I didn't consider copying in all the Quarkus dev posts from the last 5 years into the discussions as I don't know of a way to do it efficiently and with proper date. 

Got ideas ? Is it necessary ?

Reply:
It depends what problem we're trying to solve. If we don't do some sort of migration, we're trading off increased fragmentation of information (people looking for information have to look in more places, because the information might be in an old email/discussion-in-the-wrong-repo) against a better experience going forward (we get to use the github discussions interface without user-conversation contamination). Maybe that's an ok tradeoff.